src/app/clusters/basic/basic.cpp

```diff
@@ -359,7 +359,7 @@ class PlatformMgrDelegate : public DeviceLayer::PlatformManagerDelegate
             Events::ShutDown::Type event;
             EventNumber eventNumber;
 
-            CHIP_ERROR err = LogEvent(event, endpoint, eventNumber, EventOptions::Type::kUrgent);
+            CHIP_ERROR err = LogEvent(event, endpoint, eventNumber);
```

Comment:
But don't we still need to do something to flush out the shutdown event?  Or is that handled by our existing "flush events on shutdown" bits?

Reply:
i think only shutdown event needs to be urgently delivered, which can be done by ScheduleUrgentEventDeliverySync. Just updated the code a little bit, move ScheduleUrgentEventDeliverySync inside DispatchShutDownEvent in server.cpp, thanks

Reply:
We always trigger ScheduleUrgentEventDeliverySync during shutdown process.

void Engine::Shutdown()
{
    // Flush out the event buffer synchronously
    ScheduleUrgentEventDeliverySync();
....
}

Since ShutDown event is only logged during shutdown process, do we still need to call ScheduleUrgentEventDeliverySync again within  DispatchShutDownEvent in server.cpp?

Reply:
Agree, done,thanks